src/modm/driver/radio/dw3110/dw3110_phy_impl.hpp

```diff
@@ -0,0 +1,1415 @@
+	static_assert(Len == 1 || Len == 2 || Len == 4,
+				  "Masked writes only support sizes of 1,2 or 4 Bytes.");
+
+	RF_BEGIN();
+	RF_CALL(readRegister<Reg, Len, Offset>(std::span<uint8_t>(temp_rw).first<Len>()));
+	for (size_t i = 0; i < Len; i++)
+	{
+		temp_rw[i] |= or_mask[i];
+		temp_rw[i] &= and_mask[i];
+	}
+	RF_CALL(writeRegister<Reg, Len, Offset>(std::span<uint8_t>(temp_rw).first<Len>()));
+	RF_END();
+}
+
+template<typename SpiMaster, typename Cs>
+template<modm::Dw3110::Register Reg, size_t Len, size_t Offset>
+modm::ResumableResult<void>
+modm::Dw3110Phy<SpiMaster, Cs>::writeRegisterMasked(const std::span<const uint8_t, Len> or_mask,
+													const std::span<const uint8_t, Len> and_mask)
+{
+	static_assert(Len + Offset <= Reg.length,
+				  "Size of masked write is too large for this register!");
+	static_assert(Len == 1 || Len == 2 || Len == 4,
+				  "Masked writes only support sizes of 1,2 or 4 Bytes.");
+
+	RF_BEGIN();
+	RF_WAIT_UNTIL(this->acquireMaster());
+
+	tx_buffer[0] =
+		(uint8_t)(0xC0 | ((Reg.bank.addr << 1) & 0x3E) | (((Reg.offset + Offset) >> 6) & 0x01));
+	tx_buffer[1] = (uint8_t)(0x00 | ((Reg.offset + Offset) << 2));
+	if constexpr (Len == 1)
+		tx_buffer[1] |= 1;
+	else if constexpr (Len == 2)
+		tx_buffer[1] |= 2;
+	else
+		tx_buffer[1] |= 3;
+
+	Cs::setOutput(false);
+	RF_CALL(SpiMaster::transfer(tx_buffer.data(), nullptr, 2));
+	RF_CALL(SpiMaster::transfer(and_mask.data(), nullptr, Len));
+	RF_CALL(SpiMaster::transfer(or_mask.data(), nullptr, Len));
+	Cs::setOutput(true);
+
+	this->releaseMaster();
+	RF_END();
+}
+
+template<typename SpiMaster, typename Cs>
+template<modm::Dw3110::RegisterBank Reg>
+modm::ResumableResult<void>
+modm::Dw3110Phy<SpiMaster, Cs>::readRegisterBank(std::span<uint8_t> out, size_t len)
+{
+	RF_BEGIN();
+	RF_WAIT_UNTIL(this->acquireMaster());
+
+	tx_buffer[0] = (uint8_t)((Reg.addr << 1) & 0x3E);
+
+	Cs::setOutput(false);
+	RF_CALL(SpiMaster::transfer(tx_buffer.data(), nullptr, 1));
+	RF_CALL(SpiMaster::transfer(nullptr, out.data(), len));
+	Cs::setOutput(true);
+	this->releaseMaster();
+	RF_END();
+}
+
+template<typename SpiMaster, typename Cs>
+template<modm::Dw3110::RegisterBank Reg>
+modm::ResumableResult<void>
+modm::Dw3110Phy<SpiMaster, Cs>::writeRegisterBank(const std::span<const uint8_t> val, size_t len)
+{
+	RF_BEGIN();
+	RF_WAIT_UNTIL(this->acquireMaster());
+
+	tx_buffer[0] = (uint8_t)(0x80 | ((Reg.addr << 1) & 0x3E));
+
+	Cs::setOutput(false);
+	RF_CALL(SpiMaster::transfer(tx_buffer.data(), nullptr, 1));
+	RF_CALL(SpiMaster::transfer(val.data(), nullptr, len));
+	Cs::setOutput(true);
+	this->releaseMaster();
+	RF_END();
+}
+
+template<typename SpiMaster, typename Cs>
+template<size_t Len>
+bool
+modm::Dw3110Phy<SpiMaster, Cs>::checkResult(std::span<const uint8_t, Len> expected,
+											std::span<const uint8_t, Len> got)
+{
+	bool result = true;
+	for (size_t i = 0; i < Len; i++)
+	{
+		if (expected[i] != got[i]) { result = false; }
+	}
+	return result;
+}
```

Comment:
Isn't this [`std::ranges::equal()`](https://en.cppreference.com/w/cpp/algorithm/ranges/equal) ? It works on `span`s, like `std::ranges::equal(span1, span2)`. There should be no need to have this member function.